### ISSUE TITLE
Propagate TR from BIDS structure to NIfTI dims

### ISSUE THREAD

**Issue by oesteban** [nipreps/niworkflows#515]
Transfer from poldracklab/fmriprep#1093.

**Comment by effigies** [comment]
This may be obviated by running the BIDS validator first, which will error when the TR doesn't match `RepetitionTime` or `RepetitionTime` is missing.

**Comment by oesteban** [comment]
Not really, metadata can be lost (e.g., split and merge). Having this convenience tool will come handy. We're checking other things (units, dtype, xforms...).

**Comment by effigies** [comment]
I guess let's be explicit: You're saying that sometimes we find derivatives don't have pixdim4 set, but other times we do? What could it be conditional on, if not the presence of pixdim4 in the original files, which is checked by the validator? A test case would be useful.

I would also consider it problematic for splitting or merging interfaces to change or lose pixdim4. That would be a good thing to fix, as it's possible that somewhere between validation and sinking derivatives we'll have jobs that actually need the TR. We can pass the TR to some interfaces, but it is metadata that should be stored accurately in the header.
